### DISCUSSION TITLE
How to set up per object motion sensor with MQTT in HomeAssistant?

### DISCUSSION THREAD

**Discussion by goodlucknow** [blakeblackshear/frigate#5906] [Ask A Question] [answered]
Looking for advice on if its possible to set up a motion detector sensor in HA for different objects(i.e. 'person', 'car' etc).
I've searched on the topic but don't seem to be getting anywhere with it, my attempt so far below.
Anyone here have something similar already set up could set me right?

```
mqtt:
  binary_sensor:
    - name: "Front Yard Person Motion"
      device_class: motion
      state_topic: "frigate/events"
      value_template: >
        {% if (value_json['after']['camera'] == 'camera1') and (value_json['after']['label'] == 'person') and (value_json['after']['stationary'] == false) %}
          on
        {% else %}
          off
        {% endif %}
```

**Comment by goodlucknow** [comment]
I seem to have this working now, looks like the value template needed to return "ON" or "OFF", uppercase?
Code below for reference.
Would still appreciate if anyone had any insight into a better implementation, particularly regarding how to handle the off state.
Ideally I'd like it to remain "on" as long as any person is in motion. I'm pretty sure as I have it now any mqtt message on the topic not matching the template will knock it off.

```
mqtt:
  binary_sensor:
    - name: "Front Yard Person Motion"
      device_class: motion
      off_delay: 2
      state_topic: "frigate/events"
      value_template: >
        {% if (value_json['after']['camera'] == 'camera1') and (value_json['after']['label'] == 'person') and (value_json['after']['stationary'] == false) %}
          ON
        {% else %}
          OFF
        {% endif %}
```

**Comment by blakeblackshear** [comment]
The official integration creates binary sensors for each tracked object type and camera for you automatically. Is there a reason you aren't using those?

**Comment by goodlucknow** [comment]
Hi @blakeblackshear , thanks for the response!
I'm on 0.12.0-rc1, and the recommended beta integration to go with that. If that's relevant here?
My understanding of the binary sensors provided with the integration(i think) is that they provide per object "count" and "occupancy". And a general camera "motion" sensor.
The reason that doesnt work for me, and I've read previous threads where it was explained how to use the count sensor to automate lights...
I want to automate the yard light on whenever person or vehicle motion is detected.
If I park a vehicle in the drive for the night, it might be lost and found several times by frigate during the night during low light hours. So the count sensor would cause false triggers(in the context of the automation).
I think that filtering for stationary objects on my mqtt sensor means I only trigger when a desired object is in motion?

The yard light itself turning on and off could cause the cars to be detected again by frigate, but once they are not moving it wont trigger the mqtt sensor? Is my understanding correct?

**Reply by blakeblackshear** [reply]
I see. I'm not sure how you would accomplish this reliability. The automation you posted would get tripped up and turn off early when there are 2 objects and one leaves the scene.

I would suggest trying to use a zone. Another option would be to turn on the light and start a timer that expires after 10 minutes. With each mqtt message that indicates a non-stationary object is present, you could reset the timer. When the timer expires, turn off the light.

**Reply by goodlucknow** [reply]
At present I have a 5 min cooldown before the light turns off, so I think perhaps this could work reasonably well as it is. The automation is in "restart" mode, so any desired object motion retriggers it, similar to how you mention above.
If there were 2 objects and 1 left the scene, or indeed stopped moving and became stationary, I would hope the other would retrigger the mqtt sensor? Within the 5min cooldown at least.
I understand this is well outside the scope of the project, and I didn't think you'd be responding yourself, so thanks for taking the time.

**Reply by goodlucknow** [reply]
To be clear I have no issue with how frigate is working internally for recording etc. Just querying how to best use the available sensors within HA for external automations.
I haven't started working with zones yet, all on the to-do list, but frigate has been working pretty great out of the box for me so far.

**Reply by tschierer** [reply]
> 
> ```
> mqtt:
>   binary_sensor:
>     - name: "Front Yard Person Motion"
>       device_class: motion
>       off_delay: 2
>       state_topic: "frigate/events"
>       value_template: >
>         {% if (value_json['after']['camera'] == 'camera1') and (value_json['after']['label'] == 'person') and (value_json['after']['stationary'] == false) %}
>           ON
>         {% else %}
>           OFF
>         {% endif %}
> ```

@goodlucknow
How did this work out for you? Are you still using this method or did you find a better solution?
I'm currently trying to do the exact same thing, with the small area in front of our entrance. If I use the all_occupancy sensor, the light never turns off because of the parked car. If I use person_occupancy, the light turns on very late, because it's dark (hence the need for light) and the camera doesn't pick me up as a person when I get out of the car. I have to walk around the car to the entrance and when the light finally turns on, I'm already close to the door or already at the door, which sucks.
I'm also open for better solutions, if they are doable with home assistant / frigate and are reliable.

@blakeblackshear 
How would this work with a zone if the area is just about the size of a car? Is there a documentation or example for this?
I basically just need a binary sensor that is "on", if [there is a moving car or person occupancy] and is off, if [there is no person occupancy and no moving car (stationary car should not trigger the sensor)].

**Comment by edalquist** [comment, marked as answer]
I believe I just figured out an approach using triggered template sensors:

```
- trigger:
    - trigger: mqtt
      topic: frigate/events
      payload: "ON"
      value_template: |-
        {{'ON' if (
            value_json.type in ['new', 'update'] and 
            not value_json.after.stationary and 
            value_json.after.label == 'person' and 
            (value_json.after.current_zones | select('contains', 'front_lawn')|list|count > 0)
          ) else 'OFF'}}
  binary_sensor:
    - name: "Frigate Motion - Person - Front Lawn"
      unique_id: frigate_motion_front_lawn
      state: "on"
      auto_off: 60
      device_class: motion
```

This doesn't have the downside of the non-matching events from the MQTT topic turning the sensor off.

The motion timer is reset when all the following are true
* A new or update event
* A non-stationary object
* A person
* In a zone ending with __front_lawn

The template sensor will turn off 60 seconds after the event.

**Comment by auto-dgrp** [comment]
I have a similar problem, have Frigate configured correctly and working and integrated with Haas via Hacs and it is also sending MQTT topics. 
On Haas I created a conditional dashboard to show the camera image only when it detects a person in zone1. At the same time generating a visual and audible alert. 
My only problem is that the Hacs integration is not reporting the occupancy only within the zone1 and is generating many false positives. 
I read that I could solve it via MQTT, but I have never worked with it, and I'm not very skilled at working via the command line. So, while researching, I found Node Red, can anyone help me understand how to capture MQTT topics from Frigate within Node Red and generate an alert on a entity?
